framework/src/org/checkerframework/common/util/debug/GeneralAnnotatedTypeFactory.java

```diff
@@ -1,4 +1,4 @@
-package org.checkerframework.framework.type;
+package org.checkerframework.common.util.debug;
```

Comment:
I would try completely removing this class, to make sure no new uses are added.

Reply:
I don't want to rewrite `TypeOutputtingChecker` right now,  but I agree that we really don't want any new uses of `GeneralAnnotatedTypeFactory`.  I made `GeneralAnnotatedTypeFactory` a static inner class of `TypeOutputtingChecker`, so that should prevent new uses.

Reply:
That sounds like a good compromise. I would just add a comment that these two inner classes should be replaced by a normal javac implementation, to make it clear that these are historic artifacts, not interesting logic.